sklearn/datasets/openml.py

```diff
@@ -333,6 +342,41 @@ def _get_data_features(data_id, data_home):
     return json_data['data_features']['feature']
 
 
+def _get_data_qualities(data_id, data_home):
+    # OpenML API function:
+    # https://www.openml.org/api_docs#!/data/get_data_qualities_id
+    url = _DATA_QUALITIES.format(data_id)
+    error_message = "Dataset with data_id {} not found.".format(data_id)
+    json_data = _get_json_content_from_openml_api(url, error_message, True,
+                                                  data_home)
+    try:
+        return json_data['data_qualities']['quality']
+    except KeyError:
+        # the qualities might not be available, but we still try to process
+        # the data
+        return None
+
+
+def _get_data_shape(data_qualities):
+    # Using the data_info dictionary from _get_data_info_by_name to extract
+    # the number of samples / features
+    if data_qualities is None:
+        return None
+    for d in data_qualities:
```

Comment:
How about we make the code more readable/pythonic (?) by starting with `data_qualities = {d['name']: d['value'] for d in data_qualities}`?

Reply:
Ah, that's a better! (it is just a very annoying way that the data is stored .. :-))